src/Network/Network/Models/AzureFirewall/PSAzureFirewall.cs

```diff
@@ -252,21 +271,37 @@ public void RemovePublicIpAddress(PSPublicIpAddress publicIpAddress)
                 throw new ArgumentException($"Public IP Address {publicIpAddress.Id} is not attached to firewall {this.Name}");
             }
 
-            if (this.IpConfigurations.Count > 1 && ipConfigToRemove.Subnet != null)
-            {
-                throw new InvalidOperationException($"Cannot remove IpConfiguration {ipConfigToRemove.Name} because it references subnet {ipConfigToRemove.Subnet.Id}. Move the subnet reference to another IpConfiguration and try again.");
+            if (this.ManagementIpConfiguration != null) {
+                if (ipConfigToRemove.Subnet != null)
+                {
+                    Console.ForegroundColor = ConsoleColor.Yellow;
+                    Console.WriteLine($"Removing the Public IP Address associated with subnet {ipConfigToRemove.Subnet.Id}.");
+                    Console.ResetColor();
+                    ipConfigToRemove.PublicIpAddress = null;
+                }
+                else
+                {
+                    this.IpConfigurations.Remove(ipConfigToRemove);
+                }
             }
-
-            if (this.IpConfigurations.Count == 1)
+            else
             {
-                Console.ForegroundColor = ConsoleColor.Yellow;
-                Console.WriteLine($"WARNING: Removing the last Public IP Address, this will deallocate the firewall. You will have to invoke {nameof(Allocate)} to reallocate it.");
-                Console.ResetColor();
+                if (this.IpConfigurations.Count > 1 && ipConfigToRemove.Subnet != null)
+                {
+                    throw new InvalidOperationException($"Cannot remove IpConfiguration {ipConfigToRemove.Name} because it references subnet {ipConfigToRemove.Subnet.Id}. Move the subnet reference to another IpConfiguration and try again.");
+                }
 
-                this.ManagementIpConfiguration = null;
-            }
+                if (this.IpConfigurations.Count == 1)
+                {
+                    Console.ForegroundColor = ConsoleColor.Yellow;
+                    Console.WriteLine($"WARNING: Removing the last Public IP Address, this will deallocate the firewall. You will have to invoke {nameof(Allocate)} to reallocate it.");
+                    Console.ResetColor();
+
+                    this.ManagementIpConfiguration = null;
```

Comment:
No longer needed as you're in the `else` part where ManagementIpConfiguration will always be null.